lib/options.go

```diff
@@ -44,6 +45,86 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL      null.String     `json:"ttl"`
+	Strategy NullDNSStrategy `json:"strategy"`
+	// FIXME: Valid is unused and is only added to satisfy some logic in lib.Options.ForEachSpecified(),
+	// otherwise it would panic with `reflect: call of reflect.Value.Bool on zero Value`.
+	Valid bool
+}
+
+//go:generate enumer -type=DNSStrategy -transform=kebab -trimprefix DNS -output dns_strategy_gen.go
+type DNSStrategy uint8
+
+type NullDNSStrategy struct {
+	DNSStrategy
+	Valid bool
+}
+
+const (
+	DNSFirst DNSStrategy = iota + 1
+	DNSRoundRobin
+	DNSRandom
+)
+
+const DefaultDNSConfigText = "ttl=inf,strategy=first"
+
+func DefaultDNSConfig() DNSConfig {
+	return DNSConfig{
+		TTL:      null.NewString("inf", false),
+		Strategy: NullDNSStrategy{DNSFirst, false},
+	}
+}
+
+func (c DNSConfig) MarshalJSON() ([]byte, error) {
+	strat := ""
+	if c.Strategy.IsADNSStrategy() {
+		strat = c.Strategy.String()
+	}
+	return json.Marshal(struct {
+		TTL      string `json:"ttl"`
+		Strategy string `json:"strategy"`
+	}{
+		TTL:      c.TTL.String,
+		Strategy: strat,
+	})
+}
+
+func (c *DNSConfig) UnmarshalJSON(data []byte) error {
+	params := make(map[string]interface{})
+	if err := json.Unmarshal(data, &params); err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) UnmarshalText(text []byte) error {
+	if string(text) == DefaultDNSConfigText {
+		*c = DefaultDNSConfig()
+		return nil
+	}
```

Comment:
It wasn't meant as an optimization, but a way to avoid more complex logic for setting default values. I'd argue this shouldn't even be in `UnmarshalText()`, but it was one of those quick workarounds for when the exact default string is used, so the order didn't matter. I'll see if I can get rid of it.

Reply:
Yeah, so I wasn't able to remove this easily, as it causes `TestConfigConsolidation` to fail 😭 Try it for yourself, comment out this block, run the test and see all the issues... (I have an intense hatred for that specific test 😆, having seen it fail so many times. It's huge and makes it difficult to parse the errors and go to the source.)

To fix it I'd need to manually handle the default values for both fields plus their `Valid` values, which is tricky to get right without messing up some fallback logic. So I'd chalk this one up as another `TODO` for #883. Let me know if you want a comment for it or have another suggestion.